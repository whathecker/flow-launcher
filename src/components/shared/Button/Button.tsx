import React from "react";
import { Touchable, Text } from "../../Themed";
import { StyleSheet } from "react-native";
import { Container, Typography, Shadow } from "../../../styles";

type ButtonProps = {
  ctaTxt: string;
  pressHandler: () => void;
  disable?: boolean;
  fontSize?: number;
};

const Button: React.FC<ButtonProps> = ({
  ctaTxt,
  pressHandler,
  disable,
  fontSize = 18,
}: ButtonProps) => {
  const backgroundColor = disable ? "#848785" : "#0FA858";
  return (
    <Touchable
      onPress={pressHandler}
      style={styles.button}
      lightColor={backgroundColor}
      darkColor={backgroundColor}
      disabled={disable}
    >
      <Text
        style={[styles.buttonText, { fontSize: fontSize }]}
        lightColor="#fff"
        darkColor="#fff"
      >
        {ctaTxt}
      </Text>
    </Touchable>
  );
};

const styles = StyleSheet.create({
  button: {
    ...Container.centerAligned,
    width: "100%",
    paddingTop: 20,
    paddingBottom: 20,
    borderRadius: 10,
    ...Shadow.regularbackDrop,
  },
  buttonText: {
    ...Typography.p,
  },
});

export default Button;
