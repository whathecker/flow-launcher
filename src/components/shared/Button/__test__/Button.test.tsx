import * as React from "react";
import renderer from "react-test-renderer";
import Button from "../Button";

describe("Test Button component", () => {
  it("Should renders Button correctly", () => {
    const tree = renderer
      .create(
        <Button
          ctaTxt="Click it!"
          pressHandler={() => {
            return;
          }}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
