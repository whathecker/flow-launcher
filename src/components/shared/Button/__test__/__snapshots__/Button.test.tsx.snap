// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component Should renders Button correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0FA858",
      "borderRadius": 10,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 20,
      "paddingTop": 20,
      "shadowColor": "#171717",
      "shadowOffset": Object {
        "height": 5,
        "width": 3,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 3,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
        },
        Object {
          "fontFamily": "Forum-Regular",
          "fontSize": 20,
        },
      ]
    }
  >
    Click it!
  </Text>
</View>
`;
